mars/services/task/analyzer/analyzer.py

```diff
@@ -287,14 +289,23 @@ def _gen_subtask_info(
             )
             for i, fetch_chunk in zip(build_fetch_index_to_chunks, fetch_chunks):
                 inp_chunks[i] = fetch_chunk
-            copied_op = chunk.op.copy()
-            copied_op._key = chunk.op.key
-            out_chunks = [
-                c.data
-                for c in copied_op.new_chunks(
-                    inp_chunks, kws=[c.params.copy() for c in chunk.op.outputs]
-                )
-            ]
+
+            if input_changed:
+                copied_op = chunk.op.copy()
+                copied_op._key = chunk.op.key
+                out_chunks = [
+                    c.data
+                    for c in copied_op.new_chunks(
+                        inp_chunks, kws=[c.params.copy() for c in chunk.op.outputs]
+                    )
+                ]
+            else:
+                out_chunks = chunk.op.outputs
+                # Note: `dtypes`, `index_value`, and `columns_value` are lazily
+                # initialized, so we should call property `params` to initialize
+                # these fields.
+                [o.params for o in out_chunks]
```

Comment:
It's wired, what would happen without these codes?

Reply:
There will no `columns_value`, `index_value` which are used in `MainPool`.

Reply:
These field are lazily initialized, but field `a` or `b` are lazily initialized by `params`.  Can you make `a` initialized by `a`, `b` initialized by `b`? Then we can lazily initialize them in Worker Main Pool.